lib/manageiq/schema/schema_dumper.rb

```diff
@@ -66,14 +66,12 @@ def triggers(stream)
           # - Removes the "BEGIN" and "END;" stanzas
           # - Chomps the new lines from the beginning (keeps existing indent)
           # - Strips whitespace from the end (no need to worry about indent)
-          # - Indents by four spaces
           #
           # For formatting, we want to ensure at least some bit of an indent,
           # regardless of how the SQL string was originally added.
           #
           formatted_body = body.gsub(/(BEGIN$|^END;$)/, "")
                                .reverse.chomp.reverse.rstrip
-                               .indent(4)
```

Comment:
So I am not going to "put my foot down" on this change, but the reason this existed was so that this had a minimum amount of indent when dumped to the `db/schema.rb`.

Alternatively to this, you could "unindent" on the way out when executing `add_trigger` if you wanted the pgadmin diff to match.  I didn't think that postgres stored the whitespace, just the function itself (though, in retrospect, I realize I was a fool for thinking that), so I figured it didn't matter since functionally it was equivalent.

Reply:
But it does have indenting already without this extra indenting.  Removing this extra indenting is also the only value that actually makes it correct (correct in the sense that it matched what it previously in the database, which itself it also indented a little bit).  It seems the schema dumper automatically indents a little bit for presentation purposes.  This is the before/after (notice that the extra indent actually makes it extra-indented)

keeping `.indent(4)`

```ruby
  add_trigger "insteadof", "metrics", "metrics_partition", <<-SQL
          CASE EXTRACT(HOUR FROM NEW.timestamp)
            WHEN 0 THEN
              INSERT INTO metrics_00 VALUES (NEW.*);
            ...
            WHEN 23 THEN
              INSERT INTO metrics_23 VALUES (NEW.*);
          END CASE;
          RETURN NEW;
  SQL
```

removing `.indent(4)`:

```ruby
  add_trigger "insteadof", "metrics", "metrics_partition", <<-SQL
      CASE EXTRACT(HOUR FROM NEW.timestamp)
        WHEN 0 THEN
          INSERT INTO metrics_00 VALUES (NEW.*);
        ...
        WHEN 23 THEN
          INSERT INTO metrics_23 VALUES (NEW.*);
      END CASE;
      RETURN NEW;
  SQL
```

Reply:
@Fryguy it only does the indenting because there was extra indenting in the first place since we didn't use the `<~SQL` when doing the original migrations:

https://github.com/ManageIQ/manageiq-schema/blob/47979dabfb7632cd47d9ba4e7f87760af65eb0b9/db/migrate/20190122213042_use_views_for_metrics.rb#L59-L63

https://github.com/ManageIQ/manageiq-schema/blob/47979dabfb7632cd47d9ba4e7f87760af65eb0b9/db/migrate/20190122213042_use_views_for_metrics.rb#L115-L119

So this was making up for that.

Reply:
I guess what I'm thinking is I'd expect the schema dumper to preserve the original indenting, otherwise it creates a delta between what's dumped and what's in the DB.